cmake/alpakaCommon.cmake

```diff
@@ -772,7 +780,8 @@ if(TARGET alpaka)
 endif()
 
 # NVCC does not incorporate the COMPILE_OPTIONS of a target but only the CMAKE_CXX_FLAGS
-if(ALPAKA_ACC_GPU_HIP_ENABLE AND ALPAKA_CUDA_COMPILER MATCHES "nvcc")
+# For hipcc we need to propagate the CMAKE_CXX_FLAGS to HIP_HIPCC_FLAGS.
+if(ALPAKA_ACC_GPU_CUDA_ENABLE OR ALPAKA_ACC_GPU_HIP_ENABLE)
```

Comment:
How did this get here? Isn't this already part of #1324?

Reply:
You are right, this should not be part of this PR, maybe I branched from the wrong branch.

Let us decide if we set a default type or not, depending on the result I will remove this section or we close the PR.